server/src/test/java/org/apache/uniffle/server/ShuffleServerMetricsTest.java

```diff
@@ -150,7 +150,7 @@ public void testGrpcMetrics() throws Exception {
 
   @Test
   public void testServerMetricsConcurrently() throws Exception {
-    ExecutorService executorService = Executors.newFixedThreadPool(3);
+    ExecutorService executorService = Executors.newCachedThreadPool();
```

Comment:
Why do we modify this?

Reply:
IIRC, a cachedThreadPool should be faster if the number of threads started is known though this is just a test class.

Reply:
I'm not sure why these thread numbers are needed here.

Reply:
Got it.